components/automate-cli/cmd/chef-automate/cmdUtils.go

```diff
@@ -470,3 +485,91 @@ func checkResultOutputForError(output string) error {
 	}
 	return nil
 }
+
+// Execute custom command in All the node
+/*
+Input: []*NodeObject{
+
+}
+*/
+func ExecuteCmdInAllNodeAndCaptureOutput(nodeObjects []*NodeObject, singleNode bool, outputDirectory string) error {
+
+	for _, nodeObject := range nodeObjects {
+		outFiles := nodeObject.OutputFile
+		err := ExecuteCustomCmdInServiceNode(outFiles, nodeObject.InputFile, nodeObject.InputFilePrefix, nodeObject.NodeType, nodeObject.CmdString, singleNode)
+		if err != nil {
+			return err
+		}
+		if len(outFiles) > 0 {
+			if err = fileutils.Rename(outFiles[0], outputDirectory, ""); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Execute 'config show' command in specific service and fetch the output file to bastion
+func ExecuteCustomCmdInServiceNode(outputFiles []string, inputFiles []string, inputFilesPrefix string, service string, cmdString string, singleNode bool) error {
+
+	nodeMap := NewNodeTypeAndCmd()
+	fmt.Println("inputFiles: ", inputFiles)
+	cmd := ServiceNodeConstructor(nodeMap, cmdString, outputFiles, singleNode)
+	if service == POSTGRESQL {
+		cmd.CmdInputs.Args = inputFiles
+		if len(inputFiles) > 0 {
+			cmd.PreExec = prePatchCheckForPostgresqlNodes
+			cmd.CmdInputs.InputFilesPrefix = inputFilesPrefix
+		}
+	} else if service == OPENSEARCH {
+		cmd.CmdInputs.Args = inputFiles
+		if len(inputFiles) > 0 {
+			cmd.PreExec = prePatchCheckForOpensearch
+			cmd.CmdInputs.InputFilesPrefix = inputFilesPrefix
+		}
+	} else {
+		cmd.CmdInputs.Args = inputFiles
+		if len(inputFiles) > 0 {
+			cmd.PreExec = prePatchCheckForFrontendNodes
+			cmd.CmdInputs.InputFilesPrefix = inputFilesPrefix
+		}
+	}
+	switch service {
+	case AUTOMATE:
+		nodeMap.Automate = cmd
+	case CHEF_SERVER:
+		nodeMap.ChefServer = cmd
+	case POSTGRESQL:
+		nodeMap.Postgresql = cmd
+	case OPENSEARCH:
+		nodeMap.Opensearch = cmd
+	}
+	infra, err := getAutomateHAInfraDetails()
```

Comment:
getAutomateHAInfraDetails is not in util, don't use it here.

Reply:
Done.